### ISSUE TITLE
DiffPreprocPipeline_PostEddy.sh: line 375: printf: 1.793103: invalid number

### ISSUE THREAD

**Issue by MarcRamos** [Washington-University/Pipelines#24]
I installed FSL 5.0.6 but I got that error. With comma the command didn't fail. I added this in order to replace  the '.' by ','

DiffRes=`${FSLDIR}/bin/fslval ${outdir}/data/data pixdim1`
result_string="${DiffRes/./,}"
DiffRes=`printf "%0.2f" ${result_string}`

**Comment by kjamison** [comment]
It might be safer to put set the locale explicitly.  I suspect this will cause problems in other scripts as well.  Maybe in SetUpHCPPipeline.sh?  
eg:
export LC_NUMERIC="en_US.UTF-8"

**Comment by coalsont** [comment]
I agree, we should set the locale explicitly, we don't want localizations of file naming conventions, as it would make things more painful.

**Comment by coalsont** [comment]
In fact, should we just go ahead and force LANG and LC_ALL, to preempt any kind of locale strangeness from executables during scripts?

**Comment by glasserm** [comment]
I agree this should be in the setup script and otherwise am not sure what a locale is.

**Comment by coalsont** [comment]
Here is what locale environment variables do:

tim@timsdev:~$ man
What manual page do you want?
tim@timsdev:~$ LANGUAGE=fr_FR.utf8 man
Quelle page de manuel voulez-vous ?

Some languages use , for the decimal separator, such that 3/2 = "1,5".  This is the cause of the error he was getting.

**Comment by coalsont** [comment]
Here is a quick attempt at this:

https://github.com/Washington-University/Pipelines/pull/25

**Comment by coalsont** [comment]
So, I used the language that glasserm has set on his machines, but thinking a bit more about this, the en_US  locale might not be installed on some systems - I believe the C locale is guaranteed to exist, and should do roughly the same thing (sorting order is different, but that probably doesn't matter).  It is possible that an unrecognized locale may cause "C" locale behavior anyway.

Thoughts on explicitly using the C locale instead of en_US?

**Comment by glasserm** [comment]
I'm okay with that as long as it still works on my machines :P

**Comment by coalsont** [comment]
glasserm, can you test running the branch lang_override from git@github.com:coalsont/Pipelines.git on one of your machines through all steps on at least one subject?  That is:

```
git clone git@github.com:coalsont/Pipelines.git
cd Pipelines
git checkout lang_override
<copy and modify SetUpHCPPipeline.sh, leaving lang settings alone>
<run a subject through everything>
```

You really only need the modified lines in SetUpHCPPipeline.sh, so you could splice that one file into your existing setup temporarily.

**Comment by tbbrown** [comment]
Fixed by merging in pull request #25
